lib/graphql/schema.rb

```diff
@@ -57,14 +57,21 @@ class Schema
       instrument: -> (schema, type, instrumenter) { schema.instrumenters[type] << instrumenter },
       query_analyzer: ->(schema, analyzer) { schema.query_analyzers << analyzer },
       middleware: ->(schema, middleware) { schema.middleware << middleware },
+      lazy_resolve: ->(schema, lazy_class, lazy_value_method) { schema.lazy_methods.set(lazy_class, lazy_value_method) },
       rescue_from: ->(schema, err_class, &block) { schema.rescue_from(err_class, &block)}
 
     attr_accessor \
       :query, :mutation, :subscription,
       :query_execution_strategy, :mutation_execution_strategy, :subscription_execution_strategy,
       :max_depth, :max_complexity,
       :orphan_types, :directives,
-      :query_analyzers, :middleware, :instrumenters
+      :query_analyzers, :middleware, :instrumenters, :lazy_methods
+
+    class << self
+      attr_accessor :default_execution_strategy
+    end
+
+    self.default_execution_strategy = GraphQL::Execution::Execute
```

Comment:
It seems like you have been doing a lot of refactoring of SerialExecution lately.  Are you just trying to not break graphql-batch?  If so, then that leaves a lot of surface area that can break, since it only depends on SerialExecution#execute, FieldResolution#get_finished_value and FieldResolution#execution_context.strategy.  SerialExecution could even be renamed if the class is also assigned to the SerialExecution constant.